commit 65fa40276ac1318e75264e5a204547b57c2cf397
Author: Vivien Didelot <vivien.didelot@gmail.com>
Date:   Thu Apr 14 14:42:07 2016 -0400

    net: dsa: mv88e6xxx: unlock DSA and CPU ports
    
    Locking a port generates an hardware interrupt when a new SA address is
    received. This enables CPU directed learning, which is needed for 802.1X
    MAC authentication.
    
    To disable automatic learning on a port, the only configuration needed
    is to set its Port Association Vector to all zero.
    
    Clear PAV when SA learning should be disabled instead of locking a port.
    
    Fixes: 4c7ea3c0791e ("net: dsa: mv88e6xxx: disable SA learning for DSA and CPU ports")
    Signed-off-by: Vivien Didelot <vivien.didelot@savoirfairelinux.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/dsa/mv88e6xxx.c b/drivers/net/dsa/mv88e6xxx.c
index 50454be86570..028ebe2a6053 100644
--- a/drivers/net/dsa/mv88e6xxx.c
+++ b/drivers/net/dsa/mv88e6xxx.c
@@ -2478,7 +2478,7 @@ static int mv88e6xxx_setup_port(struct dsa_switch *ds, int port)
 	reg = 1 << port;
 	/* Disable learning for DSA and CPU ports */
 	if (dsa_is_cpu_port(ds, port) || dsa_is_dsa_port(ds, port))
-		reg = PORT_ASSOC_VECTOR_LOCKED_PORT;
+		reg = 0;
 
 	ret = _mv88e6xxx_reg_write(ds, REG_PORT(port), PORT_ASSOC_VECTOR, reg);
 	if (ret)